use bash when invoking dist shell scripts on solaris

binarycrusader opened the pull request:
Partially fixes #25845 

A separate, trivial fix is needed to the rust-installer scripts to completely resolve this issue.

rust-highfive commented:
Thanks for the pull request, and welcome! The Rust team is excited to review your changes, and you should hear from @alexcrichton (or someone else) soon.

If any changes to this PR are deemed necessary, please add them as extra commits. This ensures that the reviewer can see what has changed since they last reviewed the code. Due to the way GitHub handles out-of-date commits, this should also make it reasonably obvious what issues have or haven't been addressed. Large or tricky changes may require several passes of review and changes.

Please see [the contribution instructions](https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md) for more information.

alexcrichton commented:
@bors: r+

bors commented:
:pushpin: Commit 087c233 has been approved by `alexcrichton`

<!-- @bors r=alexcrichton 087c2337c156af2f7e408145737b2415e66693da -->

binarycrusader commented:
Please excuse the last commit I added; I reverted it. I'm still learning the process of GitHub's pull requests.  I didn't expect those commits to show up on this pull request.

alexcrichton commented:
@bors: r+

bors commented:
:pushpin: Commit 087c233 has been approved by `alexcrichton`

<!-- @bors r=alexcrichton 087c2337c156af2f7e408145737b2415e66693da -->
